src/write_gcm.c

```diff
@@ -3834,15 +3837,16 @@ static int wg_transmit_unique_segment(const wg_context_t *ctx,
       if (wg_format_some_of_list_ctr(ctx->resource, list, &new_list, &json,
           ctx->pretty_print_json) != 0) {
         ERROR("write_gcm: Error formatting list as JSON");
+        sfree(write_ctx.data);
```

Comment:
Recommend moving ```sfree(write_ctx.data)``` to the ```leave:``` label so you don't have to repeat it five times.

Reply:
Does the `leave:` label have access to this variable even though it's defined within the loop? If not, would it be proper to hoist the declaration to outside of the loop and re-use it in every iteration?

Reply:
Ah, I see, good point, I missed that.

Interestingly, it might be a better design to define the ```write_ctx_t``` at the very top of this function (near ```int result = -1```), and free it only at the bottom (at the ```leave:``` label)

That way you can keep reusing the response buffer for all your responses without having to keep allocating and freeing it each time through the loop. All you would have to do is set ```write_ctx.size = 0``` at the top of the ```while (list != NULL)``` loop and your buffer would be good to reuse for another round. So beautiful 😸

Reply:
Great idea! PTAL.

Reply:
+1 to declare early enough to use past the `leave:` label.

Reply:
It's beautiful but now it's in conflict with Igor's suggestion to remove write_ctx_t from the API.